datafusion/expr/src/tree_node/plan.rs

```diff
@@ -85,28 +85,25 @@ impl TreeNode for LogicalPlan {
         F: FnMut(Self) -> Result<Transformed<Self>>,
     {
         let old_children = self.inputs();
-        let t = old_children
+        let new_children = old_children
             .iter()
             .map(|&c| c.clone())
-            .map_till_continue_and_collect(f)?;
-        // TODO: Currently `assert_eq!(t.transformed, t2)` fails as
-        //  `t.transformed` quality comes from if the transformation closures fill the
-        //   field correctly.
-        //  Once we trust `t.transformed` we can remove the additional check in
-        //  `t2`.
-        let t2 = old_children
-            .into_iter()
-            .zip(t.data.iter())
-            .any(|(c1, c2)| c1 != c2);
+            .map(f)
```

Comment:
If you map all children instead of `map_till_continue_and_collect` (aka `map_until_stop_and_collect`) how will Stop work on `LogicalPlan`?

Reply:
You are right, I am reverting these. But I am a little concerned of expecting people to use `map_until_stop_and_collect` rather than `map` to activate "stop" behavior. How can we force it or set as a default behavior, can you imagine any alternative?

Reply:
Since `map_children()` is different for each `TreeNode` implementation I have no idea how else we could help developers to make "stop aware" mapping over siblings other than providing that `map_until_stop_and_collect` helper.

But I don't think that many new kinds of `TreeNode`s will be added to Datafusion in the sort or longer term future so honestly I'm less concerned.